netty/src/test/java/io/grpc/netty/NettyClientTransportTest.java

```diff
@@ -152,6 +154,26 @@ public void headersShouldOverrideDefaultUserAgent() throws Exception {
             receivedHeaders.get(USER_AGENT_KEY));
   }
 
+  @Test
+  public void maxMessageSizeShouldBeEnforced() throws Throwable {
+    startServer();
+    // Allow the response payloads of up to 1 byte.
+    NettyClientTransport transport = newTransport(newNegotiator(), 1);
+    transport.start(clientTransportListener);
+
+    try {
+      // Send a single RPC and wait for the response.
+      new Rpc(transport).halfClose().waitForResponse();
+    } catch (ExecutionException e) {
+      // Unpack the exception.
+      Throwable cause = e.getCause();
+      if (!(cause instanceof StatusException)) {
+        throw cause;
+      }
+      assertEquals(INTERNAL, ((StatusException) cause).getStatus().getCode());
```

Comment:
There are a lot of internal errors.  You might consider checking that some substring is in the error message to verify you caught the internal error you were expecting.